### PR TITLE
Use precompiled wasm images

### PR DESCRIPTION
**Pull request by mandarjog**
1. Use wasm images for stats and mx
2. Use precompiled wasm to remove initial cpu spike.

Note: This is done in anticipation of stats mx meeting performance and stability bar.
If it does not, we will change the default back to non wasm.

Will add a release note if we decide to keep the new defaults.

### CODE REVIEW

File: pilot/docker/Dockerfile.proxyv2
Comment (bianpengyuan) [inline pilot/docker/Dockerfile.proxyv2:51]: Should we just copy the pre complied ones? Original Wasm file is not used anywhere.

Comment (kyessenov) [reply pilot/docker/Dockerfile.proxyv2:51]: What's the size of these files btw?

**Comment by bianpengyuan** [comment]
Test failure are because of the removal of deprecated istio jwt filter, which the failed tests depend on. I can take over the sha update.

File: pilot/docker/Dockerfile.proxyv2
Comment (bianpengyuan) [reply pilot/docker/Dockerfile.proxyv2:51]: 5M for mx, 1.5 for stats.

Comment (kyessenov) [reply pilot/docker/Dockerfile.proxyv2:51]: I guess protobuf is 3.5M compiled :/

Comment (mandarjog) [reply pilot/docker/Dockerfile.proxyv2:51]: Keeping both in so we can continue to test easily after code freeze.
Will remove plain files later.

**Comment by bianpengyuan** [comment]
Also could you update [nullvm telemetry test ](https://github.com/istio/istio/blob/master/tests/integration/telemetry/stats/prometheus/http/nullvm/stats_filter_test.go)to explicitly set wasm enable to false to make sure they are still properly tested? like:
https://github.com/istio/istio/blob/2a7abf1162a19571faaaee58308b0f9c904646b4/tests/integration/telemetry/stats/prometheus/http/wasm/stats_wasm_filter_test.go#L46

**Comment by howardjohn** [comment]
Just wondering what is the difference from precompiled and what we have
now?

**Comment by mandarjog** [comment]
Wasm code normally get Jitted in V8, instead we package a precompiled image here.

**Comment by kyessenov** [comment]
@howardjohn It's not JITted code. v8 has to parse Wasm and this saves some of that initial translation at the expense of large code. Actual JITting is profile driven so it requires some warming.

**Comment by douglas-reid** [comment]
Since this is done in anticipation of a stability and performance bar, can we add a link / detail on the perf measurements to this PR?

**Comment by mandarjog** [comment]
folks all tests are passing, this needs to go into 1.7

**Comment by mandarjog** [comment]
@kyessenov @PiotrSikora @howardjohn

**Review by howardjohn — APPROVED** [review]
lgtm, but someone from WASM side should review the real changes so I put a hold

**Comment by howardjohn** [comment]
Nevermind, P&T is required so I did not put a hold

File: tools/packaging/deb/istio.mk
Comment (douglas-reid) [inline tools/packaging/deb/istio.mk:38]: just to be certain: our existing tests now cover this bit of VM distribution?

**Review by douglas-reid — COMMENTED** [review]
Did we ever provide a link to the perf/stability bar data?

**Review by bianpengyuan — COMMENTED** [review]
I think 1.8 bump change is already in. Do you want to update those new manifests as well? Or do you want to do it in separated PR to simplify cherry-picking?

**Comment by mandarjog** [comment]
@douglas-reid added https://github.com/istio/istio/issues/25768

@bianpengyuan is it ok to update 1.8 in a following pr?

**Comment by bianpengyuan** [comment]
@mandarjog yep, let's do it in the following PR.

File: tools/packaging/deb/istio.mk
Comment (mandarjog) [reply tools/packaging/deb/istio.mk:38]: yes, they do. That is how I found that I needed to fix this.

**Review by douglas-reid — APPROVED** [review]

**Review by kyessenov — APPROVED** [review]
It's becoming PITA to edit filter configs now that EnvoyFilters are multiplying.

**Comment by mandarjog** [comment]
aargh... all tests failed after approval.

File: tests/integration/pilot/routing_test.go
Comment (rshriram) [inline tests/integration/pilot/routing_test.go:233]: can you please revert these changes? the dot thing is nice but its masking the underlying problem.. or wait for mine to pass?

**Pull request merged**
